From fcdb5a6a5c5355c598055b2a6ae3d4268e39d8b4 Mon Sep 17 00:00:00 2001
From: Eric Harney <eharney@redhat.com>
Date: Thu, 24 Oct 2013 16:18:54 -0400
Subject: [PATCH] GlusterFS: set correct filename when cloning volume

When cloning a volume, the volume[name] field was populated
with incorrect data, resulting in an unexpected filename containing
the volume data.  This results in failures of later operations on
that cloned volume.

(cherry picked from commit bfb66019edd197141ea1462ba78dd27a2ed0e40d)

Resolves: rhbz #1023065

Upstream-Havana: https://review.openstack.org/#/c/53897/
Upstream-Icehouse: https://review.openstack.org/#/c/53735/
Launchpad Bug: #1244238
Change-Id: I067ed44cebdc8e91e9ded326953fd0c99d003f05
---
 cinder/tests/test_glusterfs.py     | 4 ++--
 cinder/volume/drivers/glusterfs.py | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/cinder/tests/test_glusterfs.py b/cinder/tests/test_glusterfs.py
index ef34fd4..44c78c9 100644
--- a/cinder/tests/test_glusterfs.py
+++ b/cinder/tests/test_glusterfs.py
@@ -600,7 +600,7 @@ class GlusterFsDriverTestCase(test.TestCase):
                                     volume_file)
         src_info_path = '%s.info' % volume_path
         volume_ref = {'id': volume['id'],
-                      'name': volume['name'] + '-clone',
+                      'name': volume['name'],
                       'status': volume['status'],
                       'provider_location': volume['provider_location'],
                       'size': volume['size']}
@@ -1508,7 +1508,7 @@ class GlusterFsDriverTestCase(test.TestCase):
                       'size': volume['size'],
                       'status': volume['status'],
                       'provider_location': volume['provider_location'],
-                      'name': 'volume-' + volume['id'] + '-clone'}
+                      'name': 'volume-' + volume['id']}
 
         drv.create_snapshot(snap_ref)
         drv._copy_volume_from_snapshot(snap_ref,
diff --git a/cinder/volume/drivers/glusterfs.py b/cinder/volume/drivers/glusterfs.py
index bd4293f..0987be0 100644
--- a/cinder/volume/drivers/glusterfs.py
+++ b/cinder/volume/drivers/glusterfs.py
@@ -162,12 +162,12 @@ class GlusterfsDriver(nfs.RemoteFsDriver):
             msg = _("Volume status must be 'available'.")
             raise exception.InvalidVolume(msg)
 
-        volume_name = CONF.volume_name_template % src_vref['id']
+        volume_name = CONF.volume_name_template % volume['id']
 
         volume_info = {'provider_location': src_vref['provider_location'],
                        'size': src_vref['size'],
                        'id': volume['id'],
-                       'name': '%s-clone' % volume_name,
+                       'name': volume_name,
                        'status': src_vref['status']}
         temp_snapshot = {'volume_name': volume_name,
                          'size': src_vref['size'],
